### PR TITLE
extend unit file specifier expansion a bit + add a test for it

### PR DESCRIPTION
**Pull request by poettering**

### CODE REVIEW

**Comment by yuwata** [comment]
lgtm.

**Comment by yuwata** [comment]
You need to list exec-specifier.service in `test/meson.build`.

File: test/test-execute/exec-specifier.service [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+ExecStart=/usr/bin/test %n = exec-specifier.service
+ExecStart=/usr/bin/test %N = exec-specifier
+ExecStart=/usr/bin/test %p = exec-specifier
+ExecStart=/usr/bin/test %P = exec/specifier
+ExecStart=/usr/bin/test %f = /exec/specifier
+ExecStart=/usr/bin/test %t = /run
+ExecStart=/usr/bin/test %S = /var/lib
+ExecStart=/usr/bin/test %C = /var/cache
+ExecStart=/usr/bin/test %L = /var/log
+ExecStart=/usr/bin/test %U = 0
```
Comment (yuwata) [inline test/test-execute/exec-specifier.service:12]: In the discussion in PR #7062, @evverx encourages to use `/bin/sh -c`. Do you agree with this?

Comment (poettering) [reply test/test-execute/exec-specifier.service:12]: not really. there are reasons to enclose things in sh, but it makes things more unreadable (and slower), hence I'd usually avoid it, really, unless there's a good reason to have it.

**Comment by poettering** [comment]
I force pushed a new and rebased version now, with the meson.build change made. Please have another look!

**Comment by yuwata** [comment]
lgtm.

**Pull request merged**